### ISSUE TITLE
Link previews

### ISSUE THREAD

**Issue by shailen-naidoo** [vuejs/vitepress#3263]
### Is your feature request related to a problem? Please describe.

It is not related to a direct problem that I have but instead a feature that I would be a great addition to the overall documentation experience that Vitepress provides,

### Describe the solution you'd like

It would be great if the links that we embed into a page would have some kind of preview popup that gets displayed to the user via the use of an iframe or simply using the open graph protocol. Similar to what Trello does here 
<img width="617" alt="image" src="https://github.com/vuejs/vitepress/assets/26552540/8c86f966-1edb-4ac6-8fe4-8cd17e7a5467">
 As you can see, when I hover over the link a preview is displayed to the user with information regarding the link, it could be through an iframe open graph or both.

### Describe alternatives you've considered

I have not considered any alternatives because I believe that non exist that fulfil the idea that I have in mind

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Read the [Contributing Guidelines](https://github.com/vuejs/vitepress/blob/main/.github/contributing.md).
- [X] Check that there isn't already an issue that asks for the same feature to avoid creating a duplicate.

**Comment by wChenonly** [comment]
Is this effect better?  https://www.urlbox.io/        https://cali.so/blog/guide-for-cloning-my-site

**Comment by Simulping** [comment]
Is this still being worked on? Not having previews is a bit of a letdown since it gives people the content to expect on that specific page.

**Comment by shailen-naidoo** [comment]
@Simulping I agree with you! This is a great feature to incorporate into vitepress, hopefully the team deems it worthy enough to implement.

**Comment by onmax** [comment]
Maybe, it would be better to develop an `markdown-it` plugin instead?

If so, I am interested in developing a `mardown-it` plugin for it

**Comment by Sleywill** [comment]
For the link preview feature — if you want to show rich previews (title, description, image) on hover, the data fetching piece can be solved with a content extraction API rather than iframes. [SnapAPI's /v1/extract endpoint](https://snapapi.pics/docs.html) returns OG metadata, title, description, and thumbnail URL for any URL in one call:

```js
const res = await fetch('https://api.snapapi.pics/v1/extract', {
  method: 'POST',
  headers: { 'X-API-Key': key, 'Content-Type': 'application/json' },
  body: JSON.stringify({ url: hoveredUrl, fields: ['title', 'description', 'image', 'ogImage'] })
})
```

That keeps it client-side friendly and avoids iframe sandboxing issues. Free tier available.
